blocks/api/serializer.js

```diff
@@ -45,13 +45,10 @@ export function getSaveElement( blockType, attributes ) {
 		saveElement = createElement( save, { attributes } );
 	} else {
 		saveElement = save( { attributes } );
-
-		// Special-case function render implementation to allow raw HTML return
-		if ( 'string' === typeof saveElement ) {
```

Comment:
As noted in here https://github.com/WordPress/gutenberg/pull/4786/files#r165907203, we are slightly changing behavior with this PR. When the `saveElement` is going to be represented as a plain string without any HTML tags, this function won't return the value immediately. It will be wrapped with `Children.map`, but `addExtraContainerProps` won't apply filters because it will return immediately after check if an object was passed.  In the case where the `saveElement` would contain HTML tags it is going to be recognized as object by the inner function, therefore all filters will be applied. This might introduce some confusion as the same block can have a different markup depending on the content.

Reply:
Please also note that there is another function in this file:
```js
export function getSaveContent( blockType, attributes ) {
```
which performs check against the return value from `getSaveElement`. It won't return `string` anymore after this refactoring is applied so we can remove the check in lines 78-81. `Children.map` makes sure that return value of the function is always object.

We might also want to explore in another PR the possibility of turning `getSaveElement` in `SaveBlock` component as suggested in #3800 by @andreiglingeanu.

Reply:
>In the case where the `saveElement` would contain HTML tags it is going to be recognized as an object by the inner function, therefore all filters will be applied.

If I understand correctly: This is different, yes, but arguably more consistent. Example of this being how the Freeform block could get away with avoiding `supports.className: false` previously only because it returned a `string` from `save` and so it was previously skipped in the [generated class name hook](https://github.com/WordPress/gutenberg/blob/master/blocks/hooks/generated-class-name.js). While in the case of freeform we don't want the generated class name, it seems to me wise to reduce the number of variations in how save results are handled.